src/main/java/uk/gov/hmcts/reform/cpo/auditlog/aop/AuditAspect.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.hmcts.reform.cpo.auditlog.aop;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.expression.AnnotatedElementKey;
+import org.springframework.expression.spel.SpelEvaluationException;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.cpo.auditlog.LogAudit;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Aspect
+@Component
+@ConditionalOnProperty(name = "audit.log.enabled", havingValue = "true")
+@Slf4j
+public class AuditAspect {
+
+    private static final String RESULT_VARIABLE = "result";
+
+    private final ExpressionEvaluator evaluator = new ExpressionEvaluator();
+
+    @Around(value = "@annotation(logAudit)")
+    public Object audit(ProceedingJoinPoint joinPoint, LogAudit logAudit) throws Throwable {
+        Object result = null;
+        try {
+            result = joinPoint.proceed();
+            return result;
+        } finally {
+            String cpoId =  getValue(joinPoint, logAudit.cpoId(), result, String.class);
```

Comment:
extra spaces here ?

Reply:
Agreed. Resolved in 8fcbda9faaefab2c281ae0d6bc0cee8f12cbcac2.